crates/ty_ide/src/completion.rs

```diff
@@ -871,6 +894,66 @@ fn is_in_definition_place(db: &dyn Db, tokens: &[Token], file: File) -> bool {
         .is_some_and(is_definition_keyword)
 }
 
+/// when the tokens indicate that the last token should be a keyword, we provide completions for that
+fn keyword_completions<'db>(tokens: &[Token]) -> Option<Completion<'db>> {
+    from_import_statement_keyword_completions(tokens)
+}
+
+/// Checks whether the tokens indicate that the last token is in the `import` section
+/// of the `import_from` definition from <https://docs.python.org/3/reference/grammar.html>.
+/// Being in this section means we match all of the parts of the definition up to `import`
+/// and we have a `Name` token in place of the `import`.
+fn from_import_statement_keyword_completions<'db>(tokens: &[Token]) -> Option<Completion<'db>> {
+    if tokens.is_empty() {
+        return None;
+    }
+
+    let from_index = tokens.iter().rposition(|t| t.kind() == TokenKind::From)?;
+
+    let tokens_after_from = &tokens[from_index + 1..];
+
+    if tokens_after_from.is_empty() {
+        return None;
+    }
+
+    // Skip over '('.' | '...')*'
+    let dots_count = tokens_after_from
+        .iter()
+        .map(Token::kind)
+        .take_while(|t| matches!(t, TokenKind::Dot | TokenKind::Ellipsis))
+        .count();
+
+    let tokens_after_dots = &tokens_after_from[dots_count..];
+
+    let (last_token, tokens_after_dots_before_last_name) = tokens_after_dots.split_last()?;
+
+    if last_token.kind() == TokenKind::Name {
+        // If we have `from .('.' | '...')* <name>` we can suggest `import`.
+        // We check that we must have at least one `.` because we don't want to suggest `import` for `from <name>`.
+        //
+        // If we have `from ('.' | '...')* dotted_name <name>` we can suggest `import`.
+        if (tokens_after_dots_before_last_name.is_empty() && dots_count > 0)
+            || is_dotted_name(tokens_after_dots_before_last_name)
+        {
+            return Some(Completion::keyword("import"));
+        }
+    }
+
+    None
+}
```

Comment:
I think this routine should be modeled on this existing function:

https://github.com/astral-sh/ruff/blob/f44598dc113a7380ca52d6c969431ca929a93473/crates/ty_ide/src/completion.rs#L667-L756

Maybe there is a way to factor out the common logic, but even if there isn't, just copying the _approach_ seems like the right move here. I agree with @MichaReiser above. And notice that the function above is careful to not do too much look-behind on the token sequence. It also handles things like ellipsis, non-logical newlines and so on.